src/ESports.ts

```diff
@@ -0,0 +1,152 @@
+import fetch from "node-fetch";
+import Discord = require("discord.js");
+import { SharedSettings } from "./SharedSettings";
+import * as CheerioAPI from "cheerio";
+import * as momentjs from "moment";
+import Botty from "./Botty";
+
+interface ESportsAPIReturnData {
+    resultsHtml: string;
+    fixturesHtml: string;
+    resultsMonths: string;
+    fixturesMonths: string;
+}
+
+interface ESportsLeagueSchedule {
+    league: string;
+    time: string;
+    teamA: string;
+    teamB: string;
+}
+
+export default class ESportsAPI {
+    private bot: Discord.Client;
+    private settings: SharedSettings;
+
+    // Map<String, Map<String, ESportsLeagueSchedule>>
+    private schedule: { [date: string]: { [league: string]: ESportsLeagueSchedule[] } } = {};
+
+    constructor(bot: Discord.Client, settings: SharedSettings) {
+        this.bot = bot;
+        this.settings = settings;
+
+        bot.on("ready", async () => {
+            await this.loadData();
+            this.postInfo();
+        });
+    }
+
+    public onCheckNext(message: Discord.Message, isAdmin: boolean, command: string, args: string[]) {
+        if (args.length !== 1) return;
+
+        // day / month
+        const formatCheck = new RegExp("\\d\\d/\\d\\d");
+        if (!formatCheck.test(args[0])) return;
+
+        const data = args[0].split("/");
+        const date = `${data[0]} ${data[1]} ${new Date().getFullYear()}`;
```

Comment:
what if the game is next year 🤔 




jk

Reply:
🤔